ntoskrnl/ntkrnlmp/CMakeLists.txt

```diff
@@ -19,35 +19,16 @@ add_executable(ntkrnlmp
     ${NTKRNLMP_PCH_SKIP_SOURCE}
     ${REACTOS_SOURCE_DIR}/ntoskrnl/ntoskrnl.rc
     ${CMAKE_CURRENT_BINARY_DIR}/ntkrnlmp.def)
-set_property(TARGET ntkrnlmp PROPERTY ENABLE_EXPORTS TRUE)
-
+set_module_type(ntkrnlmp kernel)
 target_compile_definitions(ntkrnlmp PRIVATE CONFIG_SMP)
 
-if(ARCH STREQUAL "i386")
-    set_entrypoint(ntkrnlmp KiSystemStartup 4)
-else()
-    set_entrypoint(ntkrnlmp KiSystemStartup)
-endif()
-set_subsystem(ntkrnlmp native)
-
-if(MSVC)
-    set_image_base(ntkrnlmp 0x00400000)
-    add_target_link_flags(ntkrnlmp "/SECTION:.rsrc,!DP") #Accessed from bugcheck code
-else()
-    set_image_base(ntkrnlmp 0x80800000)
-endif()
```

Comment:
Not sure it makes sense to move these things away, because there is just "one" kernel, so no need to create module type just for it. (Same comment as for the ntoskrnl cmakefile)

Reply:
If you look closely, there is already /SECTION:INIT,xxxx missing for ntkrnlmp

CMakeMacros.cmake, gcc.cmake & msvc.cmake were made to hide toolchain specifics, avoid duplicating configuration & the like, which is exactly what the PR is doing. What's your point in being so nit-picky ?